BrowserKit/Sources/Common/Extensions/URLExtension.swift

```diff
@@ -101,7 +101,7 @@ extension URL {
     /// ```
     public static func getSubdomainAndHost(from urlString: String) -> (subdomain: String?, normalizedHost: String) {
         guard let url = URL(string: urlString) else { return (nil, urlString) }
-        let normalizedHost = url.normalizedHost ?? urlString
+        let normalizedHost = url.normalizedHost?.utf8HostToAscii() ?? urlString
```

Comment:
I'm a bit confused by this PR. AFAICS, punycode support is disabled in release and only enabled in Fennec as @thatswinnie pointed out. Related commit has a Bugzilla ticket and note:

```
Bug 1312294 - Disable Punycode support until we can figure out how to prevent homographic attacks.
```

In Firefox desktop what I see when entering a URL like `https://www.xn--google-e4d.com/` is that it's translated and represented as `google.com` in the location field. But we seem to only do that for iOS in Fennec, and in fact it doesn't appear to actually be working consistently (I see google.com when tapping on the location field but not when it does not have focus, and I can't tell by the code if this is a bug or not, it seems from the code that we should be showing `google.com` in both cases, at least in Fennec).

Additionally, having taken a closer look at this, it doesn't seem like [1419275](https://bugzilla.mozilla.org/show_bug.cgi?id=1419275) is related to the change here. If needed we can create a new security ticket. But [1419275](https://bugzilla.mozilla.org/show_bug.cgi?id=1419275) appears more specific to how lengthy URLs are rendered in the location bar, and in fact is already QA validated, which I think is going to potentially create some confusion.

@PARAIPAN9 I think it would be helpful if we could do 2 things:
- Identify a more specific security ticket tracking this, or file a new one if needed that provides more clarity on what the vulnerability is and what's being fixed 
- Clarify the impact for Release. AFAICS we should not be translating punycode at all, but again LMK if I'm misunderstanding

Reply:
@mattreaganmozilla, the issue https://bugzilla.mozilla.org/show_bug.cgi?id=1419275 has evolved over time, but if you look at the older comments, you'll see the original issue that was addressed. The QA team validated a different aspect, as seen in the comments, but I will clarify this with them tomorrow. If we decide not to translate Punycode, we can clarify that, and I will close the PR.

Reply:
> the issue https://bugzilla.mozilla.org/show_bug.cgi?id=1419275 has evolved over time
> The QA team validated a different aspect, as seen in the comments, but I will clarify this with them tomorrow.

1419275 is specific to how long URLs are truncated in the location view, and punycode isn't mentioned on it. I'd really like to avoid conflating multiple issues on that Bugzilla, especially since it's already been validated. This is a source of considerable confusion from the security & QA side of things (it's often already difficult to clarify validation criteria on Bugzillas).

This PR seems like it should be referencing a Punycode-specific Bugzilla, and we can create a new one if needed. 

> If we decide not to translate Punycode

What do you mean "if we decide not to"? 🤔 It looks like Punycode translation in the location bar has been disabled since 2017.

Reply:
I'm referring to the Toolbar Redesign.

Reply:
@PARAIPAN9 LMK if you want to sync further on this, or if I can help somehow (happy to create a new Bugzilla or if you feel we should continue referencing 1419275 we can leave as-is for now)

Reply:
Thanks @mattreaganmozilla, yes, please create another ticket, it seems that this one has been "polluted" over time.

Reply:
I can stub out a new ticket but I need you to tell me the vulnerability this fixes, repro steps, and impact (we can sync if needed via Slack to avoid disclosing details on a public PR, but it needs to be clarified someplace). That's what I was trying to clarify several times above.